src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -414,6 +417,51 @@ public Void clearEntitySet( UUID entitySetId ) {
         return null;
     }
 
+    @Override
+    @RequestMapping(
+            path = { "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + ENTITY_KEY_ID_PATH + "/graph" },
+            method = RequestMethod.DELETE
+    )
+    public Void clearEntityAndNeighborEntities(
+            @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
+            @PathVariable( ENTITY_ID ) UUID entityId
+    ) {
+
+        ensureReadAccess( new AclKey( entitySetId ) );
+        Map<UUID, List<NeighborEntityDetails>> result = searchService
+                .executeEntityNeighborSearch( entitySetId, ImmutableSet.of( entityId ) );
+
+        if ( result != null && !result.containsKey( entityId ) ) {
+
+            List<NeighborEntityDetails> neighbors = result.get( entityId );
+            neighbors.parallelStream().forEach( neighbor -> {
+
+                UUID associationEntityId = (UUID) neighbor.getAssociationDetails().get( ID_FQN ).iterator().next();
+                UUID associationEntitySetId = neighbor.getAssociationEntitySet().getId();
+                Map<UUID, PropertyType> authorizedPropertyTypesOnNeighborAES = authzHelper
+                        .getAuthorizedPropertyTypes( associationEntitySetId, WRITE_PERMISSION );
+                dgm.clearEntities(
+                        neighbor.getAssociationEntitySet().getId(),
+                        ImmutableSet.of( associationEntityId ),
+                        authorizedPropertyTypesOnNeighborAES
+                );
+
+                if ( neighbor.getNeighborId().isPresent() && neighbor.getNeighborEntitySet().isPresent() ) {
+                    UUID neighborEntityId = neighbor.getNeighborId().get();
+                    UUID neighborEntitySetId = neighbor.getNeighborEntitySet().get().getId();
+                    Map<UUID, PropertyType> authorizedPropertyTypesOnNeighborES = authzHelper
+                            .getAuthorizedPropertyTypes( neighborEntitySetId, WRITE_PERMISSION );
+                    dgm.clearEntities(
+                            neighborEntitySetId,
+                            ImmutableSet.of( neighborEntityId ),
+                            authorizedPropertyTypesOnNeighborES
+                    );
+                }
+            } );
+        }
+        return null;
```

Comment:
Function name makes it sound like the entity itself will also be deleted, but it doesn't look like that's happening.